src/main/java/tatracker/model/Model.java

```diff
@@ -126,48 +107,155 @@
+     * @param editedModule the edited module {@code target}.
+     *                     The identity of {@code editedModule} must be the same as {@code target}.
      */
     void setModule(Module target, Module editedModule);
 
+    /** Returns an unmodifiable view of the filtered module list */
+    ObservableList<Module> getFilteredModuleList();
+
     /**
      * Updates the filter of the filtered module list to filter by the given {@code predicate}.
      * @throws NullPointerException if {@code predicate} is null.
      */
     void updateFilteredModuleList(Predicate<Module> predicate);
 
+    // ======== Group Methods ==================================================
+
+    /**
+     * Returns true if a given group with the same identity as {@code group}
+     * exists in a module that is in TaTracker.
+     * @param targetModule module that contains {@code group}.
+     */
+    boolean hasGroup(Group group, Module targetModule);
+
+    /**
+     * Adds the given group into a module that is in TaTracker.
+     * @param group group to add, which must not already exist in the TaTracker module.
+     * @param targetModule module to add {@code group} into, which must exist in the TaTracker.
+     */
+    void addGroup(Group group, Module targetModule);
+
+    /**
+     * Deletes the given group {@code target} from a module that is in TaTracker.
+     * @param target group to delete, which must exist in the TaTracker module.
+     * @param targetModule module to delete group {@code target} from, which must exist in the TaTracker.
+     */
+    void deleteGroup(Group target, Module targetModule);
+
+    /**
+     * Replaces the given group {@code target} in a TaTracker module with {@code editedGroup}.
+     * @param target group to edit, which must exist in the TaTracker module.
+     * @param editedGroup the edited group {@code target}.
+     *                    The identity of {@code editedGroup} must be the same as {@code target}.
+     * @param targetModule module with the group to edit, which must exist in the TaTracker.
+     */
+    void setGroup(Group target, Group editedGroup, Module targetModule);
+
+    // TODO: Group filter methods.
+
+    // ======== Student Methods ================================================
+
+    /**
+     * Returns true if a student with the same identity as {@code student} exists in the TaTracker.
+     */
+    boolean hasStudent(Student student);
+
+    /**
+     * Returns true if a given student with the same identity as {@code student}
+     * exists in a module group that is in TaTracker.
+     * @param targetGroup group to check if {@code student} is enrolled in.
+     * @param targetModule module that contains {@code group}.
+     */
+    boolean hasStudent(Student student, Group targetGroup, Module targetModule);
+
+    /**
+     * Adds the given student.
+     * {@code student} must not already exist in the ta-tracker.
+     */
+    void addStudent(Student student);
```

Comment:
Doesn't this method become redundant since we are only adding and deleting students to and from a module?